core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -223,7 +223,9 @@ public PartitionSpec apply() {
         // field IDs were not required for v1 and were assigned sequentially in each partition spec starting at 1,000.
         // to maintain consistent field ids across partition specs in v1 tables, any partition field that is removed
         // must be replaced with a null transform. null values are always allowed in partition data.
-        builder.add(field.sourceId(), field.fieldId(), field.name(), Transforms.alwaysNull());
+        // To avoid name conflict when add and remove same partition transform multiple times, field name will be
+        // replaced by field name append with field id.
+        builder.add(field.sourceId(), field.fieldId(), field.name() + "_" + field.fieldId(), Transforms.alwaysNull());
```

Comment:
I am not an expert with this part of the code, but the comment states: `field IDs were not required for v1`
Could we end up with the same `field.name() + "_" + null` if the above happens?

Maybe it would be better to add a random generated UUID instead to avoid any chance of conflicts.

Reply:
we cannot use constant name for removed field, because partition spec field name must be unique.
Since removed field is not really used, we need to change it name to some that not conflict with new field.
UUID cannot ensure generate unique.
field.fieldId() is unique, because we generate it from 1000 and auto-increment for new field.
Maybe field.name() + field.fieldId() + UUID is better?

Reply:
I know wikipedia is not every time the source of truth, but [here](https://en.wikipedia.org/wiki/Universally_unique_identifier) it states:

> A universally unique identifier (UUID) is a 128-bit label used for information in computer systems.
> [..]
> When generated according to the standard methods, UUIDs are, for practical purposes, unique."

But reading again the comment in the code, if the fieldId is always autogenerated and never `null` then we are fine with your solution as well.

Reply:
thank you for provide useful information about UUID.
and sorry about ut failed again. I think it will be work with the newest commit.